training/distributed_training/pytorch/data_parallel/maskrcnn/pytorch_smdataparallel_maskrcnn_demo.ipynb

```diff
@@ -4,26 +4,26 @@
    "cell_type": "markdown",
```

Comment:
<p>Why does "fb16" have an empty string? What does it do if parsed in the training script?</p><p>And why do you have to specify <code>data-dir</code> explicitly for SM training path?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3405/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Reply:
<p>With Python's argparse.ArgumentParser, if you specify action="store_true" for an option then it needs no value, like this:https://github.com/HerringForks/DeepLearningExamples/blob/master/PyTorch/Segmentation/MaskRCNN/pytorch/tools/train_net.py#L218.</p><p> </p><p>This notebook was not working before partly due to it is looking for the wrong data path. With FSx, the data directory will be mounted to opt/ml/input/data/train on the compute instances</p>

Reply:
On that I am actually not sure - I have only used FSx so far and never S3 directly

Reply:
<p>Ah.. well, I guess it was the training script mounting the training input path different than /opt/ml/input/data? SageMaker requires /opt/ml/input/data to properly manage training data</p>

Reply:
Yes here it was because the script was looking elsewhere and needed this override to guide it to the right location

Reply:
<p>Above this cell, please add a markdown cell:</p><p> </p><p> </p><p>In the hyperparameters dictionary, add the following:</p><p> </p><ul><li>To use AMP (FP16), add <code>"fp16": ""</code> to set <code>True</code>.In the training script, there's already an option <code>specify action="store_true"</code> that automatically set the empty inputs to <code>True</code>.</li><li>To properly set the training data path for the SageMaker training environment, add <code>"data-dir": "/opt/ml/input/data/train"</code>. </li></ul><p> </p>

Reply:
<p>BTW which one is correct between <code>data_dir</code> and <code>data-dir</code>  ? </p>

Reply:
It is the latter - these options are all defined from the script. See this [line of code](https://github.com/HerringForks/DeepLearningExamples/blob/master/PyTorch/Segmentation/MaskRCNN/pytorch/tools/train_net.py#L231)